src/main/java/org/skriptlang/skript/bukkit/tags/sources/SkriptTagSource.java

```diff
@@ -0,0 +1,55 @@
+package org.skriptlang.skript.bukkit.tags.sources;
+
+import ch.njol.util.coll.iterator.EmptyIterable;
+import org.bukkit.Keyed;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.entity.EntityType;
+import org.skriptlang.skript.bukkit.tags.TagType;
+
+public final class SkriptTagSource<T extends Keyed> extends CustomTagSource<T> {
+
+	private static SkriptTagSource<Material> ITEMS;
+	private static SkriptTagSource<Material> BLOCKS;
+	private static SkriptTagSource<EntityType> ENTITIES;
+
+	public static void makeDefaultSources() {
+		ITEMS = new SkriptTagSource<>(TagType.ITEMS);
+		BLOCKS = new SkriptTagSource<>(TagType.BLOCKS);
+		ENTITIES = new SkriptTagSource<>(TagType.ENTITIES);
+	}
+
+	/**
+	 * @param types The tag types this source will represent.
+	 */
+	@SafeVarargs
+	private SkriptTagSource(TagType<T>... types) {
+		super(TagOrigin.SKRIPT, new EmptyIterable<>(), types);
+	}
+
+	public void addTag(Tag<T> tag) {
+		tags.put(tag.getKey(), tag);
+	}
+
+	/**
+	 * @return Skript tag source for item contexts
+	 */
+	public static SkriptTagSource<Material> ITEMS() {
```

Comment:
Should these methods be named as `items`, `blocks`, `entities`?

Reply:
you tell me
i like caps but it's probably not code standards

Reply:
Isn't static final caps? Not just static

Reply:
for fields yes